lsp/lib/docs/proof_step.ml

```diff
@@ -0,0 +1,517 @@
+open Util
+open Prover
+
+(** We categorize the proof steps just to make the presentation in the UI clearer. *)
+module Kind = struct
+  type t = Module | Struct | Leaf
+
+  let to_string = function
+    | Module -> "module"
+    | Struct -> "struct"
+    | Leaf -> "leaf"
+end
+
+type t = {
+  kind : Kind.t;
+      (** Kind/Type of this proof step.
+          We want to show the proof steps differently based in its type. *)
+  status_parsed : Proof_status.t option;
+      (** Status derived at the parse time, if any.
+          This is for the omitted or error cases. *)
+  status_derived : Proof_status.t;
+      (** Derived status.
+          Here we sum-up the states from all the related obligations and sub-steps. *)
+  step_loc : Range.t;
+      (** Location of the entire step.
+          It starts with a statement/sequent and ends with the BY statement (inclusive),
+          not including the listed facts and definitions. In the case of a structured
+          proof, this ends with the BY statement of the corresponding QED step. *)
```

Comment:
I think you mean "the BY keyword", rather than "the BY statement"?

Reply:
Indeed. Thanks.